src/global.hpp

```diff
@@ -40,3 +40,21 @@
 
 #if defined(__GNUC__) && !defined(__clang__)
 #endif
+
+/*
+ * GCC-13 and GCC-14 warn about functions that take a reference and return a
+ * reference, assuming the returned reference may point into the argument, they
+ * have false positives for functions that take an id string and return a
```

Comment:
This explanation is incoherent. I don't get it.

Reply:
Better?

GCC-13 and GCC-14 warn about functions that return a reference and whose arguments also include a reference to a temporary, because they assume that the returned reference may point into the argument. This causes false positives for functions that take an id string and return a string. GCC-14 supports supressing the warnings with [[gnu::no_dangling]].

Reply:
That sounds better, though the "take and id string and return a string" is still _slightly_ confusing… I guess the point is that the input is a lookup key into a map or config, and it's returning a value stored in the map or config, right?

Though, if it's a config and not a map, this might not actually be a false positive…

Reply:
What route are you seeing to it being a true positive?

If it's a config, either it returns the attribute value owned by the config, or it returns the blank value that's statically owned by config's implementation.

Reply:
I think that makes sense as long as it's returning an attribute value and not trying to convert the attribute value to a specific type.

Reply:
Even if it is being converted, I don't see how that relates to the lifetime of the id string that's passed by reference into the function.

Reply:
It doesn't relate to the lifetime of the ID string, but if it's being converted and returning a reference, that would be bad. But the conversion functions don't return a reference so it's fine.